scio-elasticsearch/es6/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -415,32 +456,47 @@ public void setup() throws Exception {
       public void processElement(ProcessContext c) throws Exception {
         final Iterable<T> values = c.element().getValue();
 
-        // Elasticsearch throws ActionRequestValidationException if bulk request is empty,
-        // so do nothing if number of actions is zero.
-        if (!values.iterator().hasNext()) {
-          LOG.info("ElasticsearchWriter: no requests to send");
-          return;
-        }
-
         final Stream<DocWriteRequest> docWriteRequests =
             StreamSupport.stream(values.spliterator(), false)
                 .map(toDocWriteRequests::apply)
                 .flatMap(ar -> StreamSupport.stream(ar.spliterator(), false));
 
-        final Iterable<List<DocWriteRequest>> chunks =
-            Iterables.partition(docWriteRequests::iterator, maxBulkRequestSize);
+        int currentSize = 0;
+        long currentBytes = 0L;
+        ImmutableList.Builder<DocWriteRequest> chunk = ImmutableList.builder();
```

Comment:
I think `ImmutableList` is a bit unnecessary here.

Reply:
I ended up just constructing the `BulkRequest` directly,